packages/concerto-core/lib/decoratormanager.js

```diff
@@ -253,6 +253,233 @@ class DecoratorManager {
         return newModelManager;
     }
 
+    /**
+     * Extracts all the decorator commands from all the models in modelManager
+     * @param {ModelManager} modelManager the input model manager
+     * @param {object} options - decorator models options
+     * @param {boolean} options.removeDecoratorsFromModel - flag to strip out decorators from models
+     * @param {string} options.locale - locale for extracted vocabulary set
+     * @returns {Object} a new model manager with the decorations removed and a list of extracted decorator jsons
```

Comment:
If it's returning a new `ModelManager`, then the return type is `ModelManager` is it not?

Reply:
It is returning an Object of type:
```
{
            modelManager:newModelManager,
            decoratorCommandSet,
            vocabularies
}
```